components/notifications-service/integration_test/notification_test.go

```diff
@@ -0,0 +1,1207 @@
+		require.NoError(t, err)
+
+		postData := ts.GetLastPost()
+		assert.NotNil(t, postData)
+		assert.True(t, postData.BasicAuthUsed)
+		assert.Equal(t, "integration_test_username_secretstore", postData.BasicAuthUsername)
+		assert.Equal(t, "integration_test_password_secretstore", postData.BasicAuthPassword)
+	})
+}
+
+func TestNotificationDispatchWithOneWebhookAlertRuleForAllEvents(t *testing.T) {
+	err := suite.DeleteEverything()
+	require.NoError(t, err)
+
+	ts := newTestServer()
+	defer ts.Close()
+
+	// NOTE: the elixir notifications service caches the rules and refreshes them
+	// every 5s, which means we have to wait that long for the rules to take
+	// effect before we test their behavior. To speed things up, we group tests
+	// that can share the same alert rules state.
+	addRules(t,
+		webhookAlertRuleFor(api.Rule_CCRFailure, ts.WebhookURL()),
+		webhookAlertRuleFor(api.Rule_ComplianceFailure, ts.WebhookURL()),
+		// Success events are unsupported, which we test for below
+		webhookAlertRuleFor(api.Rule_ComplianceSuccess, ts.WebhookURL()),
+		webhookAlertRuleFor(api.Rule_CCRSuccess, ts.WebhookURL()),
+	)
+
+	t.Run("CCRFailure alerts are sent to the Webhook URL", func(t *testing.T) {
+		_, err := suite.Client.Notify(ctx, ccrFailureEvent())
+		require.NoError(t, err)
+		postData := ts.GetLastPost()
+		assert.NotNil(t, postData)
+	})
+
+	t.Run("CCRFailure alerts are formatted properly", func(t *testing.T) {
+		_, err := suite.Client.Notify(ctx, ccrFailureEvent())
+		require.NoError(t, err)
+		postData := ts.GetLastPost()
+		assert.NotNil(t, postData)
+
+		// uncomment to see the message the service sent
+		//fmt.Println(string(postData.BodyBytes))
+
+		var actual WebhookMessage
+		err = json.Unmarshal(postData.BodyBytes, &actual)
+		require.NoError(t, err)
+
+		assert.Equal(t, "converge_failure", actual.Type)
+		assert.Equal(t, "chefnode.example", actual.NodeName)
+
+		expectedSnippet := "Chef client run failure on [https://automate.example/nodes/0] chefnode.example : https://automate.example/nodes/0/runs/1\nExample Exception Occurs\nEXAMPLE EXCEPTION MESSAGE\n(cookbook_example::recipe_example line 13) \n"
+		assert.Equal(t, expectedSnippet, actual.FailureSnippet)
+		assert.Equal(t, "Example Exception Occurs", actual.ExceptionTitle)
+		assert.Equal(t, "EXAMPLE EXCEPTION MESSAGE\n(cookbook_example::recipe_example line 13)", actual.ExceptionMessage)
+
+		expectedBT := []string{
+			"/path/to/recipe_example.rb: 42",
+			"/path/to/recipe_example.rb: 23",
+		}
+		assert.Equal(t, expectedBT, actual.ExceptionBacktrace)
+		assert.NotEmpty(t, actual.AutomateFQDN)
+		assert.Equal(t, "https://automate.example/nodes/0/runs/1", actual.AutomateFailureURL)
+
+		actualStartTime, err := time.Parse(time.RFC3339Nano, actual.StartTimeUTC)
+		assert.NoError(t, err)
+		assert.Equal(t, startTime, actualStartTime)
+
+		actualEndTime, err := time.Parse(time.RFC3339Nano, actual.EndTimeUTC)
+		assert.NoError(t, err)
+		assert.Equal(t, endTime, actualEndTime)
```

Comment:
No time stamp check needed for this test?

Reply:
There are many subtle differences between the different formats associated with the different rule types. This one doesn't have the timestamp field: https://github.com/chef/automate/blob/22545b9d39d4a575e21e7c738a63076fb178cff6/components/notifications-service/server/lib/formatters/webhook.compliance.ex